include/avif/avif.h

```diff
@@ -48,6 +48,21 @@ extern "C" {
 #define AVIF_API
 #endif // defined(AVIF_DLL)
 
+#if defined(AVIF_ENABLE_NODISCARD) || (defined(__cplusplus) && __cplusplus >= 201700L) || \
```

Comment:
Vincent: Why do you compare `__cplusplus` with `201700L` here?

If we are trying to detect C++17, we should compare  `__cplusplus` with `201703L`. See 

- https://en.wikipedia.org/wiki/C%2B%2B17
- https://en.cppreference.com/w/cpp/preprocessor/replace#Predefined_macros
- https://learn.microsoft.com/en-us/cpp/build/reference/zc-cplusplus?view=msvc-170.

Reply:
I found that the comparison with `201700L` was copied to libwebp: https://chromium-review.googlesource.com/c/webm/libwebp/+/5020699

Reply:
Indeed, bad copy/paste. I fixed it there too.

Reply:
Fixed in https://github.com/AOMediaCodec/libavif/pull/2246